python/tchannel/tornado/connection.py

```diff
@@ -1,18 +1,25 @@
 from __future__ import absolute_import
 
 import functools
+import logging
+import os
+import socket
+import sys
+
+from tornado import gen
+from tornado import iostream
 
 from .. import frame
 from .. import messages
 from .. import exceptions
 from ..io import BytesIO
 from ..context import Context
+from ..messages import CallResponseMessage
 from ..messages.types import Types
 from ..messages.common import PROTOCOL_VERSION
 
-from tornado import gen
-from tornado.iostream import StreamClosedError
-from tchannel.messages import CallResponseMessage
+
+log = logging.getLogger('tchannel')
 
 
 class TornadoConnection(object):
```

Comment:
shall we make two subclass of Tornadoconnection.  One is for outgoingconnection and the other one is for inboundconnection. both can implement their own logic and share the common part